tools/src/main/java/com/nedap/archie/flattener/FlattenerConfiguration.java

```diff
@@ -0,0 +1,144 @@
+package com.nedap.archie.flattener;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.nedap.archie.util.KryoUtil;
+
+public class FlattenerConfiguration {
+
+    /**
+     * If true, create an operational template
+     * default false
+     */
+    private boolean createOperationalTemplate = false;
+    /**
+     * If true, replaces the archetype root with a C_COMPLEX_OBJECT with the archetype_ref set
+     * If false, leaves the ArchetypeRoot in place, with the children as attributes, since archetype root itself is a C_COMPLEX_OBJECT
+     * Default false
+     */
+    private boolean useComplexObjectForArchetypeSlotReplacement = false;
+    /**
+     * Remove all zero occurrences objects. Default false, except for OperationalTemplates
+     */
+    private boolean removeZeroOccurrencesObjects = false;
+    /**
+     * Remove zero occurrences objects in parents. Useful for modeling tools, where you cannot undo a zero occurrences if it's defined in a parent, so don't need to know that it exists
+     */
+    private boolean removeZeroOccurrencesInParents = false;
+    /**
+     * Set to true to remove languages from the metadata and translations. Only for operational templates. Default false
+     */
+    private boolean removeLanguagesFromMetaData = false;
+    /**
+     * The languages to not remove. Only works when removeLanguagesFromMetadata=true && createOperationalTemplate == true
+     */
+    private String[] languagesToKeep = null;
+
+    /**
+     * Only for Operational templates: replace the use node constructs with their replacements. Default true.
+     */
+    private boolean replaceUseNode = true;
+
+    /**
+     * Only for Operational templates: replace the archetype roots with the corresponding structure. Defaults to true
+     */
+    private boolean fillArchetypeRoots = true;
+    /**
+     * Only for Operational templates: remove any closed archetype slots from the archetype.
+     */
+    private boolean closeArchetypeSlots = true;
+
+    public static FlattenerConfiguration forFlattened() {
+        return new FlattenerConfiguration();
+    }
+
+    public static FlattenerConfiguration forOperationalTemplate() {
+        FlattenerConfiguration result = new FlattenerConfiguration();
+        result.setCreateOperationalTemplate(true);
+        result.setRemoveZeroOccurrencesObjects(true);
```

Comment:
This is currently not used. Is that intentional?

Reply:
It is intentional. The current interface is new Flattener().createOperationalTemplate(). However, since it now got a new constructor with a configuration, it must be possible also to create a Flattener that creates operational templates like this. so this is part of the public API

A test would be nice, but not right now.